third-party/realdds/include/realdds/topics/notification/notification-msg.h

```diff
@@ -62,116 +68,37 @@ class notification
-    };
-
-    struct video_stream_profile
-    {
-        int8_t stream_index;     // Used to distinguish similar streams like IR L / R
-        int16_t uid;             // Stream unique ID
-        int16_t framerate;       // FPS
-        int8_t format;           // Corresponds to rs2_format
-        int8_t type;             // Corresponds to rs2_stream
-        int16_t width;           // Resolution width [pixels]
-        int16_t height;          // Resolution width [pixels]
-        bool default_profile;    // Is default stream of the sensor
-        //intrinsics - TODO
-    };
+    static bool take_next( dds_topic_reader &,
+                           notification * output,
+                           eprosima::fastdds::dds::SampleInfo * optional_info = nullptr );
 
-    struct motion_stream_profile
-    {
-        int8_t stream_index;     // Used to distinguish similar streams like IR L / R
-        int16_t uid;             // Stream unique ID
-        int16_t framerate;       // FPS
-        int8_t format;           // Corresponds to rs2_format
-        int8_t type;             // Corresponds to rs2_stream
-        bool default_profile;    // Is default stream of the sensor
-    };
 
-    struct video_stream_profiles_msg
-    {
-        char group_name[32];     // Streams can be grouped together to indicate physical or logical connection
-        size_t num_of_profiles;  
-        video_stream_profile profiles[MAX_VIDEO_PROFILES];
-    };
-
-    struct motion_stream_profiles_msg
-    {
-        char group_name[32];     // Streams can be grouped together to indicate physical or logical connection
-        size_t num_of_profiles; 
-        motion_stream_profile profiles[MAX_MOTION_PROFILES];
-    };
-
-    enum class control_result : uint16_t
-    {
-        SUCCESSFUL,
-        FAILED,
-        MAX_CONTROL_RESULT  // TODO - for success/failure bool is enough. Consider how to pass exception information
-    };
-
-    struct control_response_msg
-    {
-        uint32_t message_id;  // Running counter
-        uint32_t response_to; // message_id of the control this message is responding to
-        control_result result;
-        //TODO - double value for options or vector of bytes for HW monitor
-    };
+    notification() = default;
+    notification( raw::notification && );
+    notification( nlohmann::json const &, uint32_t version = 0 );
+    notification( data_type type, nlohmann::json const &, uint32_t version = 0 );
 
-#pragma pack( pop )
+    nlohmann::json json_data() const;
 
-    static std::string construct_topic_name( const std::string& topic_root )
-    {
-        return topic_root + "/notification";
-    }
-
-    template<typename T>
-    static void construct_raw_message( msg_type msg_id, const T& msg, raw::device::notification& raw_msg )
+    // Get the custom data with casting to the desired type
+    // Syntax: auto stream_info = msg.get< STREAM_INFO >();
+    //         if ( msg ) do something with it;
```

Comment:
Not sure I understand. What is `get`? Shouldn't it be
// Syntax: auto stream_info = msg.custom_data< STREAM_INFO >();
//         if ( stream_info ) do something with it;

Reply:
Copy paste issue - `get` was renamed to `custom_data`. I'll update.

Reply:
Fixed.